opnsense.logstash.conf

```diff
@@ -27,6 +27,23 @@ filter {
     }
     if [syslog_program] == "filterlog" {
       opnsensefilter { field_name => "syslog_message" }
+      geoip {
+        source => "source"
+      }
+    }
+    if [syslog_program] == "suricata" {
+      if [src_ip]  {
+        geoip {
+          source => "src_ip"
+        }
+        if ![geoip.ip] {
+          if [dest_ip]  {
+            geoip {
+              source => "dest_ip"
```

Comment:
Isn't dest_ip the IP of the server?

Reply:
Might be an RFC1918 IP, where GeoIP makes no sense

Reply:
geoip returns NULL if the source parameter is an RFC1918 address. No "miss" is logged. So I'm not worried about that.

Good catch on the dest_ip. That config only works on my use case, analysing inbound traffic.

Reply:
After a proper [https://www.elastic.co/guide/en/logstash/current/plugins-filters-geoip.html#plugins-filters-geoip-source](reRTFM) I remembered why I did this. 'source' is a required field, while 'target' is optional.

The parameter names are not self-explanatory. 'geoip.source' might be the destination or source IP, it doesn't matter. It has to be not NULL though in order for geoip to do anything.

My assumption here is that one of the fields may contain an public routeable IP.
If it's not the src_ip, it may be the dest_ip. If both are RFC1918, well, geoip fails silently.

I missed the case where both fields are !RFC1918. And I need to fix the filterlog...

Reply:
No rush, cannot test until next week and because of that I cannot merge it anyway till then.